tenable/io/v3/vm/exclusions/schema.py

```diff
@@ -0,0 +1,101 @@
+'''
+Exclusion API Endpoint Schemas
+'''
+
+from datetime import datetime
+
+from marshmallow import Schema, fields, validate
+from marshmallow.decorators import post_dump, pre_load, validates_schema
+
+
+class RulesSchema(Schema):
+    '''
+    Schema for rrules field for ScheduleSchema class
+    '''
+    freq = fields.Str(
+        validate=validate.OneOf(
+            ['ONETIME', 'DAILY', 'WEEKLY', 'MONTHLY', 'YEARLY']
+        ),
+    )
+    interval = fields.Int()
+    byweekday = fields.List(
+        fields.Str(validate=validate.OneOf(
+            ['SU', 'MO', 'TU', 'WE', 'TH', 'FR', 'SA'])
+        )
+    )
+    bymonthday = fields.Int(
+        validate=validate.OneOf(list(range(1, 32)))
+    )
+
+    @pre_load
+    def pre_serialization(self, data, **kwargs):
+        if 'freq' in data:
+            data['freq'] = str(data['freq']).upper()
+
+        if data.get('freq') == 'WEEKLY':
+            if 'byweekday' not in data:
+                data['byweekday'] = ['SU', 'MO', 'TU', 'WE', 'TH', 'FR', 'SA']
+            data['byweekday'] = [str(i).upper() for i in data['byweekday']]
+            data.pop('bymonthday', None)
+        elif data.get('freq') == 'MONTHLY':
+            if 'bymonthday' not in data:
+                data['bymonthday'] = datetime.today().day
+            data.pop('byweekday', None)
+        else:
+            data.pop('bymonthday', None)
+            data.pop('byweekday', None)
+        return data
+
+    @post_dump
+    def post_serialization(self, data, **kwargs):
+        if 'byweekday' in data:
+            if data['byweekday']:
+                data['byweekday'] = ','.join(data['byweekday'])
+
+        return data
+
+
+class ScheduleSchema(Schema):
+    '''
+    Schema for schedule field for AgentExclusionSchema class
+    '''
+    enabled = fields.Boolean()
+    starttime = fields.DateTime(format='%Y-%m-%d %H:%M:%S')
+    endtime = fields.DateTime(format='%Y-%m-%d %H:%M:%S')
```

Comment:
Correct me if I am wrong! Do we have to follow ISO-8601 format? As I didn't seen on https://confluence.eng.tenable.com/display/PORG/%5BSDK%5D+Exclusions as well

Reply:
You can ignore this comment if there is no requirement for IOS-8601

Reply:
@alpesh-te As per our discussion with @tomeara In the request body, we don't need to use ISO8601 format for starttime and endtime. but we are validating the date format as per v2 standard.

Reply:
Thanks for information, we will take care  ISO8601 date format with integration testing